python/cuml/test/test_pickle.py

```diff
@@ -18,50 +18,41 @@
 import pickle
 import pytest
 
-from cuml.test.utils import array_equal, unit_param, stress_param
+from cuml.test.utils import array_equal, unit_param, stress_param, \
+    ModuleConfig
 from cuml.test.test_svm import compare_svm
 from sklearn.base import clone
-from sklearn.datasets import load_iris, make_classification,\
-    make_regression
+from sklearn.datasets import load_iris, make_classification, make_regression
 from sklearn.manifold.t_sne import trustworthiness
 from sklearn.model_selection import train_test_split
 
-regression_models = {
-    "LinearRegression": lambda fit_intercept=True: cuml.LinearRegression(
-        fit_intercept=fit_intercept),
-    "LogisticRegression": lambda fit_intercept=True: cuml.LogisticRegression(
-        fit_intercept=fit_intercept),
-    "Lasso": lambda fit_intercept=True: cuml.Lasso(
-        fit_intercept=fit_intercept),
-    "Ridge": lambda fit_intercept=True: cuml.Ridge(
-        fit_intercept=fit_intercept),
-    "ElasticNet": lambda fit_intercept=True: cuml.ElasticNet(
-        fit_intercept=fit_intercept)
-}
+regression_config = ModuleConfig(module=cuml.linear_model)
+regression_models = regression_config.get_models()
 
-solver_models = {
-    "CD": lambda: cuml.CD(),
-    "SGD": lambda: cuml.SGD(eta0=0.005),
-    "QN": lambda: cuml.QN(loss="softmax")
-}
+solver_config = ModuleConfig(
```

Comment:
How can we ensure that all models are picked up even when someone adds a new module? Like if "cuml.perceptron" shows up someday, could it slip through and add models that are not tested for pickling? Could we run `ModuleConfig(cuml)` and recursively find all classes?

Reply:
Absolutely, something like 
```python
def get_modules_from_package(package):
    return inspect.getmembers(package, inspect.ismodule)
```
would work. Working on it!